src/OCServo.h

```diff
@@ -0,0 +1,89 @@
+#ifndef OCSERVO_H
+#define OCSERVO_H
+
+#include <Arduino.h>
+#include <SoftwareSerial.h>
+
+#include "ROBSMotor.h"
+
+enum OCSMode {
+    OCS_SERVO,
+    OCS_MOTOR
+};
+
+enum OCSResponseLevel {
+    OCS_RESPOND_READ_ONLY,
+    OCS_RESPOND_ALL
+};
+
+typedef enum e_SerialPortType {
+    SERIAL_HARDWARE,
+    SERIAL_SOFTWARE
+} SerialPortType;
+
+struct OCSResponse {
+    byte prefix[2];
+    byte id;
+    byte length;
+    byte instruction;
+    int numberOfParameters;
+    byte parameters[16];
+    byte checksum;
+};
+
+class OCServo {
+    private:
+        byte id;
+        Stream *serial;
+        SerialPortType portType;
+
+        byte getChecksum(byte *data, int size);
+        byte baudRateToByte(long baudrate);
+        long byteToBaudRate(byte value);
+        OCSResponse bytesToResponse(byte *data, int size);
+        OCSResponse readResponse();
```

Comment:
- Add brief comments to each method declaration would help users understand their purpose and intended usage.
 - Add const qualifiers to methods that don’t modify the class’s state, which improves safety and signals to the user that the function will not alter the object.


```suggestion
        uint8_t calculateChecksum(const uint8_t *data, size_t size) const;
        uint8_t baudRateToByte(uint32_t baudRate) const;
        uint32_t byteToBaudRate(uint8_t value) const;
        OCSResponse parseResponse(const uint8_t *data, size_t size) const;
        OCSResponse readResponse() const;
```